src/index.ts

```diff
@@ -0,0 +1,124 @@
+
+import Ajv from 'ajv';
+import addFormats from 'ajv-formats';
+import ajvErrors from 'ajv-errors';
+import { jsonSchemaV3 } from './json-schema-v3';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const toJsonSchema = require('@openapi-contrib/openapi-schema-to-json-schema');
+
+import type { SchemaParser, ValidateSchemaInput, ParseSchemaInput, SchemaValidateResult } from '@asyncapi/parser';
+import type { v2 } from '@asyncapi/parser/esm/spec-types';
+import type { ValidateFunction, ErrorObject } from 'ajv';
+
+export function OpenAPISchemaParser(): SchemaParser {
+  return {
+    validate,
+    parse,
+    getMimeTypes,
+  };
+}
+export default OpenAPISchemaParser;
+
+async function validate(input: ValidateSchemaInput<unknown, unknown>): Promise<SchemaValidateResult[]> {
+  const validator = getAjvInstance().getSchema('openapi') as ValidateFunction;
+
+  let result: SchemaValidateResult[] = [];
+  const valid = validator(input.data);
+  if (!valid && validator.errors) {
+    result = ajvToSpectralResult(input.path, [...validator.errors]);
+  }
+
+  return result;
+}
+
+async function parse(input: ParseSchemaInput<unknown, unknown>): Promise<v2.SchemaObject> {
```

Comment:
```suggestion
async function parse(input: ParseSchemaInput<unknown, unknown>): Promise<AsyncAPISchema> {
```

Reply:
I know it perfectly well 😂 I wait for your PR.

Reply:
self service! https://github.com/asyncapi/parser-js/pull/656 :D